### ISSUE TITLE
[Debug only] CodeActionEditHandlerService has long running work on UI thread that makes debug bits unusable.

### ISSUE THREAD

**Issue by balajikris** [dotnet/roslyn#1737]
See [here](http://source.roslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/Implementation/CodeActions/CodeActionEditHandlerService.cs,100) and [here](http://source.roslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/Implementation/CodeActions/CodeActionEditHandlerService.cs,156) . These go through every single document in the solution and gets diagnostics while committing a lightbulb item. This makes debug bits completely unusable because this is on the UI thread and VS becomes unresponsive. 

``` C#
public void Apply(...)
...
#if DEBUG
            var documentErrorLookup = new HashSet<DocumentId>();
            foreach (var project in workspace.CurrentSolution.Projects)
            {
                foreach (var document in project.Documents)
                {
                    if (!document.HasAnyErrors(cancellationToken).WaitAndGetResult(cancellationToken))
                    {
                        documentErrorLookup.Add(document.Id);
                    }
                }
            }
#endif
...
#if DEBUG
            foreach (var project in workspace.CurrentSolution.Projects)
            {
                foreach (var document in project.Documents)
                {
                    if (documentErrorLookup.Contains(document.Id))
                    {
                        document.VerifyNoErrorsAsync("CodeAction introduced error in error-free code", cancellationToken).Wait(cancellationToken);
                    }
                }
            }
#endif
...
}
```

**Comment by srivatsn** [comment]
The check itself is useful but may be it could run in the background.

**Comment by shyamnamboodiripad** [comment]
This just tripped me again. As part of a bug fix, I had to make some changes around cancellation tokens and how they are used for 'fix all' suggested action in the light bulb. While testing my fix in the RoslynDev hive, I noticed wait dialog coming up when I wasn't expecting it to (i.e. my change should not have done anything to cause wait dialog to come up).

After spending bunch of time debugging, it turned out the above slow debug only code was the thing responsible for causing the wait dialog to be displayed and that the issue had nothing to do with my cancellation token changes.

Fixing so that we don't run into this again would be a good idea.
